test.js

```diff
@@ -447,3 +450,60 @@ test('alias - require previousId and userId', t => {
     })
   })
 })
+
+test.cb('e2e', t => {
+  const id = uid(16)
+
+  const testFn = function () {
+    var axiosClient = axios.create({
+      baseURL: 'https://api.runscope.com',
+      timeout: 10 * 1000,
+      headers: { 'Authorization': 'Bearer ' + process.env.RUNSCOPE_TOKEN }
+    })
+
+    retries(axiosClient, { retries: 3 })
+
+    axiosClient.get('buckets/zfte7jmy76oz/messages?count=10')
+        .then(function (response) {
+          t.is(response.status, 200)
+
+          // TODO: fail and end the test if none of the requests match.
+          // Currently the test will just hang forever if this happens, because
+          // we never call t.end() in this case.
+          response.data.data.forEach(item => {
+            axiosClient.get('buckets/zfte7jmy76oz/messages/' + item.uuid)
+              .then(function (response) {
+                t.is(response.status, 200)
+
+                const body = JSON.parse(response.data.data.request.body)
+
+                if (id === body.properties.id) {
+                  t.pass()
+                  t.end()
+                }
+              })
+              .catch(function (error) {
+                t.fail(error)
+                t.end()
+              })
+          })
+        })
+        .catch(function (error) {
+          t.fail(error)
+          t.end()
+        })
+  }
+
+  // https://segment.com/segment-libraries/sources/analytics_node_e2e_test/overview
+  const analytics = new Analytics('wZqHyttfRO0KxEHyRTujWZQswgTDZx1N', { flushAt: 1 })
+  analytics.track({
+    userId: 'prateek',
+    event: 'E2E Test',
+    properties: {
+      id: `${id}`
+    }
+  }, function () {
+    setTimeout(testFn, 5000)
+  })
+  analytics.flush()
```

Comment:
Do we need this `.flush()`? It looks like we tell the library to flush after a single call.

Reply:
Probably not, I didn't totally trust the library to do the right thing here so I added both.

Reply:
> I didn't totally trust the library

That's not good. Let's chat about that on Slack.

Maybe you should remove it to gain trust in the library? 😉

Reply:
Yep I can change that.

Reply:
I left the flush but removed the flushAt:1 part. Can reverse that if we want.